### DISCUSSION TITLE
Different disks sizes and number best practices

### DISCUSSION THREAD

**Discussion by EdoaLive** [minio/minio#16012] [Q&A] [answered]
Hello,
I'm preparing a 5-node deployment but since the hardware is from different providers, the disks have the same storage space, but differently distributed.

- 3 nodes have 2x4TB
- 2 nodes have 4x2TB

My initial thought was to join 2 of the 4 disks with LVS or MD to give minio the same "view" on all nodes, but doing so I'm adding a layer and taking away the direct access to disks on these nodes. 
Is it better to give all disks to every node (having different sets)? How would the final total space result?
What is the best practice here?

**Comment by harshavardhana** [comment, marked as answer]
If you use different disk sizes in a same cluster, your capacity is governed by the the smallest disk in the setup.

**Reply by EdoaLive** [reply]
So is it better to join them via LVM or MD? JBOD or RAID0?
Thanks

**Reply by klauspost** [reply]
No, it is better to get uniformly sized disks. Anything you put above pure disks will just make your cluster less predictable and often slower.
